src/hooks/useResultCallback.ts

```diff
@@ -1,19 +1,104 @@
+import { pipe } from "@effect/data/Function"
+import * as Hash from "@effect/data/Hash"
 import * as Effect from "@effect/io/Effect"
-import * as Exit from "@effect/io/Exit"
-import * as Fiber from "@effect/io/Fiber"
+import type * as Fiber from "@effect/io/Fiber"
 import * as Ref from "@effect/io/Ref"
 import * as Runtime from "@effect/io/Runtime"
 import * as Stream from "@effect/stream/Stream"
 import type { ResultBag } from "effect-react/hooks/useResultBag"
-import { updateNext, useResultBag } from "effect-react/hooks/useResultBag"
+import { initialTrackedProps, makeResultBag, updateTrackedProps } from "effect-react/hooks/useResultBag"
 import type { RuntimeContext } from "effect-react/internal/runtimeContext"
 import * as Result from "effect-react/Result"
-import { useCallback, useContext, useEffect, useRef, useState } from "react"
+import type { DependencyList } from "react"
+import { useCallback, useContext, useRef, useSyncExternalStore } from "react"
 
-type FiberState<E> = { readonly _tag: "Idle" } | {
-  readonly _tag: "Running"
-  readonly fiber: Fiber.RuntimeFiber<E, void>
-  readonly interruptingRef: Ref.Ref<boolean>
+class FiberStore<R, E, A> {
+  constructor(
+    readonly runtime: Runtime.Runtime<R>
+  ) {}
+
+  listeners: Array<() => void> = []
+
+  subscribe = (listener: () => void) => {
+    this.listeners.push(listener)
+
+    return () => {
+      this.listeners.splice(this.listeners.indexOf(listener), 1)
+
+      queueMicrotask(() => {
+        if (this.listeners.length === 0) {
+          this.interruptIfRunning()
+        }
+      })
+    }
+  }
```

Comment:
maybe we should just leverage the [native EventTarget](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/EventTarget) for this. For 3 reasons:
1. It implements the required behavior in the native code layer
2. it [adheres to abort signals](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/removeEventListener#:~:text=Event%20listeners%20can%20also%20be%20removed%20by%20passing%20an%20AbortSignal%20to%20an%20addEventListener()%20and%20then%20later%20calling%20abort()%20on%20the%20controller%20owning%20the%20signal.), making it a possible target for more complex effect operations
3. you have much greater visibility on listerners since they show up in the listener registery. Making listener-leaks and other such bugs much easier to find and understand

it doesn't matter if you implement an event target inside the class, or have the class be an event target, I don't mind either way

Reply:
Had a look at that web api, it seems it performs worse than a simple loop. What was the api for detecting if an event is no longer being listened to?

Reply:
for _detecting_? not aware that's possible in general.. funny it works worse than a loop. If you find it less-suitible then the current impl is fine 

if you're talking about the 3rd point, idk take a look at the Event Listeners tab in the browser devtools, also they can be identified in the sources tab, where you can place breakpoints for listeners directly

Reply:
You can simulate it just fine
```ts
class MyTarget extends EventTarget {
  amount = 0

  constructor() {
    super()
    super.addEventListener("cleanup", this.interruptIfRunning)
  }

  addEventListener(type: string, callback: EventListenerOrEventListenerObject | null, options?: boolean | AddEventListenerOptions | undefined): void {
      this.amount++
    super.addEventListener(type, callback, options)
  }

  removeEventListener(type: string, callback: EventListenerOrEventListenerObject | null, options?: boolean | EventListenerOptions | undefined): void {
      this.amount--
    super.removeEventListener(type,callback,options)
  }
  
  // dispatch event can use some CustomEvent extends Event type thing
  // and so on and so forth
}
```
But theres no real need to force it if we don't see a clear benefit

Reply:
I'm in the "lets don't force it" camp at the moment :)